### PR TITLE
fix Issue 23622 - ImportC #defines conflict with declarations

### PR DESCRIPTION
**Pull request by WalterBright**
If they conflict, the declaration wins.

### CODE REVIEW

File: compiler/src/dmd/dsymbol.d [hunk 1]
```diff
@@ -2620,6 +2620,16 @@ Dsymbol handleSymbolRedeclarations(ref Scope sc, Dsymbol s, Dsymbol s2, ScopeDsy
...
     auto vd = s.isVarDeclaration(); // new declaration
     auto vd2 = s2.isVarDeclaration(); // existing declaration
+
+    if (vd && vd.isCmacro())
+        return vd2;
+
+    if (vd2 && vd2.isCmacro())
+    {
+        sds.symtab.update(vd);      // replace vd2 with vd
+        return vd;
```
Comment (dkorpel) [inline compiler/src/dmd/dsymbol.d:2630]: This branch is not tested

Comment (WalterBright) [reply compiler/src/dmd/dsymbol.d]: replaced with assert

**Comment by WalterBright** [comment]
This is blocking several linux standard .h files from compiling.

**Review by thewilsonator — APPROVED** [review]

**Comment by WalterBright** [comment]
Thanks @thewilsonator !

**Comment by WalterBright** [comment]
All the C Standard .h files that Ubuntu has now smoke test with ImportC.

**Pull request merged**